pages/api/user/avatar.ts

```diff
@@ -0,0 +1,73 @@
+import crypto from "crypto";
+import type { GetServerSidePropsContext, NextApiRequest, NextApiResponse } from "next";
+
+import { getSession } from "@lib/auth";
+import prisma from "@lib/prisma";
+import { defaultAvatarSrc } from "@lib/profile";
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const username = req.url?.substring(1, req.url.lastIndexOf("/"));
+  const user = await prisma.user.findUnique({
+    where: {
+      username: username,
+    },
+    select: {
+      avatar: true,
+      email: true,
+    },
+  });
+
+  const emailMd5 = crypto
+    .createHash("md5")
+    .update(user?.email as string)
+    .digest("hex");
+  const img = user?.avatar;
+  if (img) {
+    const decoded = img
+      .toString()
+      .replace("data:image/png;base64,", "")
+      .replace("data:image/jpeg;base64,", "");
+    const imageResp = Buffer.from(decoded, "base64");
+    res.writeHead(200, {
+      "Content-Type": "image/png",
+      "Content-Length": imageResp.length,
+    });
+    res.end(imageResp);
+  } else {
+    res.writeHead(302, {
+      Location: defaultAvatarSrc({ md5: emailMd5 }),
+    });
+    res.end();
+  }
+}
+
+export async function getServerSideProps(context: GetServerSidePropsContext) {
```

Comment:
Can we user `getServerSideProps` on `/api` routes? 🤔

Reply:
I believe not; doubtful this does anything.

Reply:
Good catch. I had forgotten about this bit. Yeah, it doesn't do anything.